pkg/windows/windows.go

```diff
@@ -191,12 +193,30 @@ func New(workerIgnitionEndpoint, vxlanPort string, instanceInfo *instance.Info,
 		return nil, errors.Wrapf(err, "unable to setup VM %s sshConnectivity", instanceInfo.Address)
 	}
 
+	// When in powershell, this will evaluate to an object containing information about the shell.
+	// When in cmd.exe, this will not evaluate to anything, and will be used literally. This can be used to determine
+	// if cmd or Powershell is being used.
+	powershellVersionTable := "($PSVersionTable)"
+	out, err := conn.run("echo " + powershellVersionTable)
+	if err != nil {
+		return nil, errors.Wrap(err, "error determining default shell")
+	}
+	var powershellDefault bool
+	if strings.TrimSpace(out) == powershellVersionTable {
+		powershellDefault = false
+	} else if strings.Contains(out, "PSEdition") {
+		powershellDefault = true
+	} else {
+		return nil, errors.Wrapf(err, "unexpected output when determining default shell: %s", out)
```

Comment:
In case the default shell is other than cmd or powershell, should WMCO default to powershell instead? If so, why don't always default to powershell?

Reply:
In general defaulting to powershell would not be the way to go, if there would be a default it would using the `powershell -Command` wrapper. But I'd prefer the default case to error out early, and prevent undefined behavior later, which should reduce the amount of hard to solve bugs :)